### DISCUSSION TITLE
get list of all project branches via api/command line

### DISCUSSION THREAD

**Discussion by N6REJ** [community/community#23776] [New to GitHub]
<p>I’m trying to figure out how I can get a list of remote branches without creating a local repository. For example.</p>
<p>I want to pull a list of all branches at <a href="https://github.com/joomla/joomla-cms.git" rel="nofollow noopener">https://github.com/joomla/joomla-cms.git</a><br>
go something like git ls-branches <a href="https://github.com/joomla/joomla-cms.git" rel="nofollow noopener">https://github.com/joomla/joomla-cms.git</a> would be ideal.</p>
<p>I will be doing this via either CLI or php dependening on which one will work.</p>
<p>either way each time the script is call the url could be different.</p>
<p>Thanks in advance</p>

**Comment by AndreaGriffiths11** [comment, marked as answer]
<p>Hi <a class="mention" href="/n6rej">@n6rej</a>,</p>
<p>Thanks for being here! I’m sharing the solution you’ve received from Support here as it might help others. </p>
<p>There are a few ways that you can do this. I’d recommend from the terminal, and using v3 of our API (REST) you can make a GET request to fetch these specifically (as documented here: <a href="https://developer.github.com/v3/repos/branches/#list-branches" rel="nofollow noopener">https://developer.github.com/v3/repos/branches/#list-branches</a>)</p>
<pre><code>curl https://api.github.com/repos/joomla/joomla-cms/branches
</code></pre>
<p>Or if this is a private repository that you have access to then you’ll need to make an authenticated request using either an OAuth / Personal Access Token or using basic authentications (username / password) (as documented here:<a href="https://developer.github.com/v3/#authentication" rel="nofollow noopener">https://developer.github.com/v3/#authentication</a>):</p>
<pre><code>curl -H "Authorization: token $TOKEN" https://api.github.com/repos/joomla/joomla-cms/branches
</code></pre>
<p>In the above request you’d substitute the placeholder $TOKEN with your access token.</p>
<p>If you’d like to achieve this via PHP there are a number of libraries that wrap around this API, for instance: <a href="https://github.com/KnpLabs/php-github-api" rel="nofollow noopener">https://github.com/KnpLabs/php-github-api</a> however this is a third party API so we cannot support issues with the libraries implementation directly (you’ll have to raise an issue on the repository itself).</p>

**Reply by azzarb** [reply]
Hi @AndreaGriffiths11 ,

How can I get a specific branch and list all the files inside it ..is there any curl command for that

**Comment by N6REJ** [comment]
<p>TY!</p>
<p>I did stumble upon a way that hopefully is ok with staff.</p>
<pre><code>for /f "tokens=3* delims=/" %%f in ('call git ls-remote --heads %repo%') do (

REM check to be sure we got a list of branches back.
	IF ERRORLEVEL 1 (
    	ECHO Sorry we couldn't find any branches in the %repo%
    	EXIT /B
	) ELSE (

	REM yep, we did so lets populate the array of branches.
	set /a "count=count+=1"

	REM Populate array
	set branch[!count!]=%%f

	)
)
</code></pre>
<p>I literally searched for days to figure SOMETHING out.  I really appreciate your help.</p>

**Comment by chiyang10000** [comment]
<p>Hi <a class="mention" href="/andreagriffiths11">@andreagriffiths11</a> </p>
<p>I have successfully recieved reponse from the github branches api.</p>
<p>However, the response did’nt contain all the branches in my repo but 10 of 100.</p>
<p>How could I get all my branches info?</p>
<p>Looking forward to hearing from you.</p>

**Comment by gauravdalal** [comment]
<p>This is <strong>super accurate solution</strong> to the problem (for CLI).<br>
Thanks!</p>
